docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+* has `Component` and `Gate` within the templates.
+* does not depend on any of the other three components.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Circuit Commands (Dash) <a name="circ-comd"></a>
+
+This section provides details on the implementation of the various electronic circuit commands.
+
+There are 4 different types of components of electronic circuits that can be instantiated in the program:
+* `Resistor` - A _resistor_ component.
+* `Capacitor` - A _capacitor_ component.
+* `Inductor` - An _inductor_ component.
+* `VoltageSource` - An _Alternating Current_ Voltage Source.
+
+Each component is used within a circuit template, based on the kind of circuit instantiated.
+
+<div markdown="span" class="alert alert-primary">
+
+:information_source: **Note:** The `VoltageSource` is instantiated in all Circuit Templates.
+
+</div>
+
+There are four different circuit templates that can be instantiated in the program:
+
+* `LcTemplate` - Inductor-Capacitor Circuit Template
+* `RTemplate` - Resistor Circuit Template
+* `RcTemplate` - Resistor-Capacitor Template (extends `RTemplate`)
+* `LrTemplate` - Inductor-Resistor Template (extends `RTemplate`)
+
+![CircuitCommandClass](diagrams/CircuitCommandClassDiagram.png)
+
+<small><i>Figure 6</i></small>
+
+The diagram above demonstrates the relationship between the various `CircuitCommand` objects. The various commands to be parsed are as explained in this section. While the **User Guide** explains the commands used on the CLI, this section goes into detail the classes used to execute the commands. The command classes also make use of the classes in the [**`Model`**](#model-component) shown in the diagram below when executing the commands.
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
+
+### `TemplateCircuitCommand`  <a name="temp-circ"></a>
+Given below is the sequence diagram for interactions within the `logic` and `template` components for the 
+`parse(template rl)` API call that implements the `template` command to create templates.
+
+![TemplateSequence](diagrams/TemplateSequenceDiagram.png)
+
+<small><i>Figure 8</i></small>
+
+### `SetCircuitCommand` <a name="set-circ"></a>
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("set r 500")` API call which implements the `set` command to set values of components.
+
+![SetSequence](diagrams/SetSequenceDiagram.png)
+
+<small><i>Figure 9</i></small>  
+
+### `AddCircuitCommand` <a name="add-circ"></a>
+
+Given below is the detailed Sequence diagram for interactions within the `logic`, `template` and `component` components 
+for the `parse("add parallel c 20")` API call that implements the `add` command to add components.
+
+![AddSequence](diagrams/AddSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/20852776/97526728-bbdd1c00-19e4-11eb-8c55-b7db4d29cce9.png)

try breaking the diagram into smaller fragments using ref frames?

Reply:
I really like this suggestion since I wasn't aware of it, thanks! Should be part of https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/121.